src/net/if_.rs

```diff
@@ -26,6 +26,22 @@ pub fn if_nametoindex<P: ?Sized + NixPath>(name: &P) -> Result<c_uint> {
     }
 }
 
+/// Resolve an interface number into an interface.
+pub fn if_indextoname(index: c_uint) -> Result<CString> {
+    // We need to allocate this anyway, so doing it directly is faster.
+    let mut buf = vec![0u8; IF_NAMESIZE];
+
+    let return_buf = unsafe {
+        libc::if_indextoname(index, buf.as_mut_ptr() as *mut _)
+    };
+
+    if return_buf.is_null() {
+        Err(Error::last())
+    } else {
+        Ok(CStr::from_bytes_until_nul(buf.as_slice()).unwrap().to_owned())
+    }
```

Comment:
Nix has a nice helper function to help you handle those common syscall error cases:

```suggestion
   nix::errno::Errno::result(return_buf)?;
   Ok(CStr::from_bytes_until_nul(buf.as_slice()).unwrap().to_owned())
```

Reply:
I had to add a `.cast()` to the `return_buf`, since `ErrnoSentinel` is only implemented for `*mut c_void`.